src/kbmod/regionsearch/utilities.py

```diff
@@ -0,0 +1,316 @@
+        colnames : typing.List[str]
+            The list of column names to read from the file. The file may contain other columns but it must have these columns.
+        Returns
+        -------
+        bool
+            True if the table was read and the version is valid and all of the columns are present.
+            False if the table was not read or the version is invalid or any of the columns are missing.
+        """
+        hold_version = self.version
+        try:
+            table = astropy.table.Table.read(filename, format=format)
+            for metakey in table.meta.keys():
+                setattr(self, metakey, table.meta[metakey])
+            if self.version != hold_version:
+                self.version = hold_version
+                raise Exception(f"Version mismatch: {self.version} != {table.meta['version']}")
+            for column in colnames:
+                if column not in table.colnames:
+                    raise Exception(f"Column {column} not found in {filename}")
+                setattr(self, column, table[column])
+        finally:
+            self.version = hold_version
+        return True
+
+    def write_table(self, filename: str, format: str, colnames: typing.List[str]):
+        """
+        Write the table to the file including al the columns in colnames.
+
+        Parameters
+        ----------
+        filename : str
+            The full filename.
+        format : str
+            The format of the file to read or write.
+        colnames : typing.List[str]
+            The list of column names to write to the file.
+        """
+        if os.path.exists(filename):
+            os.remove(filename)
+        with open(filename, "w") as f:
+            coldata = [getattr(self, column) for column in colnames]
+            table = astropy.table.Table(
+                data=coldata,
+                names=colnames,
+            )
+            table.meta["version"] = self.version
+            table.write(f, format=format, overwrite=True)
+        return True
+
+    def _generate_data(self):
+        """
+        Generate the data for the test.
+        """
+        if self.data_loaded:
+            return
+
+        random.seed(self.seed)
+        # timerange to assign to samples. The time of a sample is not related to the cluster or sample in cluster.
+        self.timerange = time.Time(
+            ["2022-06-01T00:00:00.000", "2023-06-01T00:00:00.000"], format="isot", scale="utc"
+        )
+        self.timerange.format = "mjd"
+        # the nominal distances for each cluster.
+        if self.clustercnt <= 1:
+            self.clusterdistances = [random.randrange(2, 10000)]
+        else:
+            self.clusterdistances = [2]
+            self.clusterdistances.extend([random.randrange(2, 10000) for _ in range(self.clustercnt - 1)])
+
+        clusters = [[random.uniform(0.0, 360.0), random.uniform(0, 180.0)] for _ in range(self.clustercnt)]
```

Comment:
Side note: In the limit this will create an over density around the poles. It doesn't matter in this case because the number of clusters will likely be small.